lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java

```diff
@@ -731,4 +734,81 @@ public void onChannelStateChanged(ChannelStateChange stateChange) {
             assertEquals("Suspended channel histories do not match", suspendedChannelHistory, expectedSuspendedChannelHistory);
         }
     }
+
+    /**
+     * <p>
+     * Verifies that the {@code ConnectionManager} enters the disconnected state and sets the suspend timer
+     * upon unavailable transport.
+     * </p>
+     * <p>
+     * Spec: RTN15g
+     * </p>
+     */
+    @Test
+    public void connection_manager_enters_disconnected_state_on_transport_failure() throws AblyException, NoSuchFieldException, IllegalAccessException {
+        ClientOptions opts = createOptions(testVars.keys[0].keyStr);
+        try(AblyRealtime ably = new AblyRealtime(opts)) {
+            ConnectionManager connectionManager = ably.connection.connectionManager;
+            connectionManager.connect();
+            new Helpers.ConnectionManagerWaiter(ably.connection.connectionManager).waitFor(ConnectionState.connected);
+
+            // Here, we "fake" being online for 2 minutes - the suspendTime is set by onConnected and the default is 2 minutes
+            Field suspendTimeField = connectionManager.getClass().getDeclaredField("suspendTime");
+            suspendTimeField.setAccessible(true);
+            suspendTimeField.set(connectionManager, System.currentTimeMillis() - 10);
+
+            // We also have to grab the "real" transport to pass the superseded test
+            Field transportField = connectionManager.getClass().getDeclaredField("transport");
+            transportField.setAccessible(true);
+
+            connectionManager.onTransportUnavailable((ITransport) transportField.get(connectionManager), new ErrorInfo());
+            new Helpers.ConnectionManagerWaiter(connectionManager).waitFor(ConnectionState.disconnected);
+
+            assertTrue((long) suspendTimeField.get(connectionManager) >= System.currentTimeMillis());
+
```

Comment:
Hmm, is that to assert that the suspend time will have reset after disconnection and that was the reason to set suspend time -10 in the first place?

Reply:
Yes - this is asserting that `onTransportUnavailable` will reset the `suspendTime` :)